Edit an event for admin

RReiso opened the pull request:
This PR allows admin to edit the meetup through an HTML form.

Issue: Admin: edit meetup, (https://github.com/wnbrb/wnb-rb-site/issues/24)
 - [x] Creating a Admin::EventsController#update controller action that edits an event with the provided params
- [x] This action should validate that there is a current_user and redirect to admin/login if no user is logged in.
- [x] If the event with the provided ID does not exist, the controller should redirect to the 404 page
- [x] Creating a Rails view with a form where an admin user can update and save the event information
 If the provided information does not create a valid event, errors should be displayed on the corresponding form fields or in a banner at the top of the page.
 
Done in collaboration with @sawiggins

egiurleo commented on app/controllers/admin/events_controller.rb:
```diff
@@ -9,5 +9,45 @@ def index
     end
+
+    def edit
+      redirect_to new_user_session_path unless current_user
```
You can accomplish the same thing here using Pundit's DSL: `authorize Event` (check out line 7 in this file and PR #52)

What you wrote here is totally functional, I just want to make sure we're consistent in how we do things!

egiurleo commented on app/controllers/admin/events_controller.rb:
```diff
@@ -9,5 +9,45 @@ def index
+    end
+
+    def update
+      redirect_to new_user_session_path unless current_user
```
Same comment here!

egiurleo reviewed (commented):
All of the functionality here looks _awesome_ -- I tested it manually and it works great too. I left one small comment about using Pundit for authorization, and I would also appreciate if you could add some tests (there are already some very similar tests in `spec/controllers/admin/events_controller_spec`).

@RReiso I know you said you'd be busy with bootcamp, so please let me know if I can jump in and help with any of this, esp because it took me so long to review.

KaylaGallatin commented:
One small note: Now that https://github.com/wnbrb/wnb-rb-site/pull/61 has been merged we could update the edit links on the index page to link to this new edit page (or get this in a follow-up PR).

RReiso commented:
@egiurleo,
Me and @sawiggins have created event_controller tests for #edit and #update actions. 
We also updated the authorising of the event with Pundit and added the links to edit_event pages as per @KaylaGallatin note. After updating the event, we changed the redirection from admin_dashboard to admin_events page.
Thanks, Emily

egiurleo commented on spec/controllers/admin/events_controller_spec.rb:
```diff
@@ -20,6 +20,10 @@
         get :index
         expect(response).to have_http_status(401)
       end
+      it 'returns 401' do
+        get :index
+        expect(response).to have_http_status(401)
+      end
```
Looks like this test was duplicated!

RReiso replied:
Thanks, Emily. Duplicate test deleted!

egiurleo reviewed (commented):
Okay this is looking excellent, except there is a [duplicated test](https://github.com/wnbrb/wnb-rb-site/pull/62/files#diff-66da4e17c976b2975b6455792f1aa46bdbd2bc9305b32b05d77a75553d90e7cdR23) in the admin/events_controller spec. If you delete that test I will approve and merge :D

egiurleo reviewed (approved):
🎉 awesome job!
